pkg/cmd/server/api/v1/conversions.go

```diff
@@ -149,6 +151,18 @@ func SetDefaults_NodeConfig(obj *NodeConfig) {
 	}
 	SetDefaults_ClientConnectionOverrides(obj.MasterClientConnectionOverrides)
 
+	if len(obj.DeprecatedNodeIP) != 0 {
+		if obj.KubeletArguments == nil {
+			obj.KubeletArguments = ExtendedArguments{}
+		}
+
+		if _, ok := obj.KubeletArguments["node-ip"]; ok {
+			glog.Warning("nodeIP config is ignored in favor of kubeletArguments[\"node-ip\"]")
```

Comment:
don't print in defaulting/conversion... make it a warning or error in config validation

Reply:
also, it's only a warning/error if they mismatch